### PR TITLE
simple solution checker

### PR DESCRIPTION
**Pull request by vitornesello**

### CODE REVIEW

File: src/solution_check.jl [hunk 1]
```diff
@@ -0,0 +1,105 @@
+function check_tw_bounds(tws::Vector{Range}, target_time::Float64)
+    
+    for tw in tws
+        if target_time <= tw.ub && target_time >= tw.lb
+            return true
+        end
+    end
+    return false
+
+end
```
Comment (IssamT) [inline src/solution_check.jl]: The function has nothing specific about time windows. therefore generic names would be more appropriate

File: src/solution_check.jl [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                        ongoing_req_ids::Set{String})
+
+    for act in route.sequence
+        if act.request_id != "" # Part of a request
+            req = data.requests[computed_data.request_id_2_index[act.request_id]]
+            if req.request_type == 0
+                if act.operation_type == 1
+                    if in(req.id, complete_req_ids)
+                        error("Request ", req.id, " is performed more than once.")
+                    end
```
Comment (IssamT) [inline src/solution_check.jl]: Why is the above test nested. It should be done just after line 19
https://github.com/atoptima/RichVehicleRoutingProblem.jl/pull/32/files#diff-bca8ce6d51b53a49a7c6ff082335b8a7R19

File: src/solution_check.jl [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                elseif act.operation_type == 2
+                    if in(req.id, ongoing_req_ids) == false
+                        error("Request ", req.id, " is delivered before being picked-up.")
+                    end
+                    push!(complete_req_ids, req.id)
+                end
+            else # request_type in [1, 2]
+                if in(req.id, complete_req_ids)
+                    error("Request ", req.id, " is performed more than once.")
+                end
```
Comment (IssamT) [inline src/solution_check.jl]: This test won't be needed. refer to the previous comment

File: src/solution_check.jl [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                push!(complete_req_ids, req.id)
```
Comment (IssamT) [inline src/solution_check.jl]: We still need to check that operation_type is correct

File: src/solution_check.jl [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                        solution::RvrpSolution)
+
+    nb_used_vehicles = Dict{String,Int}(vehicle_set.id => 0 for vehicle_set in data.vehicle_sets)
+    complete_req_ids = Set{String}()
+    ongoing_req_ids = Set{String}()
+
+    for r in solution.routes
+        v_set = data.vehicle_sets[computed_data.vehicle_set_id_2_index[r.vehicle_set_id]]
+        nb_used_vehicles[v_set.id] += 1
+        if nb_used_vehicles[v_set.id] > v_set.nb_of_vehicles_range.soft_range.ub
```
Comment (IssamT) [inline src/solution_check.jl]: The if condition is incomplete

File: src/solution_check.jl (hunk 2 above)
Comment (vitornesello) [reply src/solution_check.jl]: I did not understand. What test?

File: src/solution_check.jl [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+            # Check properties
+            for (k,v) in product_specification.property_requirements
+                if v < vehicle_properties[k]
+                    error("In action ", act.id, " of route ", route.id,
+                          ": Uses property (", k, ") that vehicle ",
+                          v_set.id, " does not have. Vehicle properties: ",
+                          vehicle_properties, ".")
+                end
+            end
+            if act.operation_type == 1 || act.operation_type == 2
```
Comment (IssamT) [inline src/solution_check.jl]: This condition is not needed as it is already tested in line 37 https://github.com/atoptima/RichVehicleRoutingProblem.jl/pull/32/files#diff-bca8ce6d51b53a49a7c6ff082335b8a7R37

**Pull request merged**